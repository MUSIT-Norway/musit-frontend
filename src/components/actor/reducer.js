import { Observable } from 'rxjs/Rx';
import actions from "./actions";

const dataFromServer = [
  {
    id: 1,
    fn: 'Jarl parl'
  },
  {
    id: 2,
    fn: 'Line pine'
  },
  {
    id: 3,
    fn: 'Kjell pell'
  },
  {
    id: 4,
    fn: 'Stein sein'
  },
  {
    id: 5,
    fn: 'Knut lut'
  },
  {
    id: 6,
    fn: 'Ellen'
  }
]

const initialState = [];

const ActorReducer$ = Observable.of(() => initialState)
  .merge(
    actions.suggest$.map((update) => () => {
      if (!update) {
        return initialState;
      }
      return dataFromServer.filter(d => {
        const name = d.fn.toLowerCase();
        const query = update.value.toLowerCase();
        return name.indexOf(query) > -1
      })
    }),
    actions.clear$.map(() => () => initialState),
  );

export default Observable.merge(
  ActorReducer$.map(reducer => ["suggest", reducer]),
);