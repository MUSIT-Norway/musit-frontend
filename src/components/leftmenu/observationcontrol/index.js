
import React, { Component, PropTypes } from 'react'
import { Button, Row, Col } from 'react-bootstrap'
import FontAwesome from 'react-fontawesome'

export default class ObservationControlComponent extends Component {
  static propTypes = {
    id: PropTypes.number.isRequired,
    translate: PropTypes.func.isRequired,
    onClickNewObservation: PropTypes.func.isRequired,
    onClickNewControl: PropTypes.func.isRequired
  }

  render() {
    const { id, translate, onClickNewControl, onClickNewObservation } = this.props
    const getTranslate = (term) => (translate(`musit.leftMenu.observationControl.${term}`))
    const buttonLogic = (type, eventType) => {
      return (
        <Button
          id={`${id}_${type}`}
          onClick={(event) => eventType(event.target.value)}
          style={{ textAlign: 'left', width: '100%' }}
        >
          <FontAwesome name="plus-circle" style={{ padding: '2px' }} />
          {getTranslate(type)}
        </Button>
      )
    }
    return (
      <div>
        <Row>
          <Col xs={6} sm={12}>
            {buttonLogic('newObservation', onClickNewObservation)}
          </Col>
          <Col xs={6} sm={12}>
            {buttonLogic('newControl', onClickNewControl)}
          </Col>
        </Row>
      </div>
    )
  }
}
