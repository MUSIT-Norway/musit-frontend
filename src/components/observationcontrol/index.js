import React from 'react';
import ObservationControlGrid from './Grid';
import ObservationControlComponent from './LeftMenu';
import Layout from '../../layout';
import Breadcrumb from '../../layout/Breadcrumb';
import Toolbar from '../../layout/Toolbar';
import { hashHistory } from 'react-router';

export default class ObservationControlGridShow extends React.Component {
  static propTypes = {
    translate: React.PropTypes.func.isRequired,
    observationControlGridData: React.PropTypes.arrayOf(React.PropTypes.object),
    params: React.PropTypes.object,
    route: React.PropTypes.object,
    loadActorDetails: React.PropTypes.func.isRequired,
    loadControlAndObservations: React.PropTypes.func.isRequired,
    loadStorageObj: React.PropTypes.func.isRequired,
    path: React.PropTypes.arrayOf(React.PropTypes.object)
  }

  constructor(props) {
    super(props);
    this.props.params.id = this.props.params.id * 1;
    this.state = {
      showObservations: this.props.route.showObservations,
      showControls: this.props.route.showControls
    };
  }

  componentWillMount() {
    this.props.loadControlAndObservations(this.props.params.id, {
      onSuccess: (result) => {
        if (result && result.length > 0) {
          this.props.loadActorDetails({data: result.map(r => r.doneBy)});
        }
      }
    });
    this.props.loadStorageObj(this.props.params.id);
  }

  makeToolbar() {
    return <Toolbar
      showRight={this.state.showControls}
      showLeft={this.state.showObservations}
      labelRight="Kontroller"
      labelLeft="Observasjoner"
      placeHolderSearch="Filtrer i liste"
      clickShowRight={() => this.setState({ ...this.state, showControls: !this.state.showControls })}
      clickShowLeft={() => this.setState({ ...this.state, showObservations: !this.state.showObservations })}
    />;
  }

  makeLeftMenu() {
    return <div style={{ paddingTop: 10 }}>
      <ObservationControlComponent
        id={this.props.params.id}
        translate={this.props.translate}
        selectObservation
        selectControl
        onClickNewObservation={() => hashHistory.push(`/magasin/${this.props.params.id}/observation/add`)}
        onClickNewControl={() => hashHistory.push(`/magasin/${this.props.params.id}/control/add`)}
      />
    </div>;
  }

  makeContent() {
    return <ObservationControlGrid
      id={this.props.params.id}
      translate={this.props.translate}
      tableData={this.props.observationControlGridData.filter((e) => {
        if (e.eventType && this.state.showControls && this.state.showObservations) {
          return true;
        } else if (e.eventType && this.state.showControls) {
          return e.eventType.toLowerCase() === 'control';
        } else if (e.eventType && this.state.showObservations) {
          return e.eventType.toLowerCase() === 'observation';
        }
        return false;
      })}
    />;
  }

  render() {
    return (
      <Layout
        title="Magasin"
        translate={this.props.translate}
        breadcrumb={
          <Breadcrumb
            node={this.props.rootNode}
            onClickCrumb={(node) => hashHistory.push(node.url)}
            allActive
          />
        }
        toolbar={this.makeToolbar()}
        leftMenu={this.makeLeftMenu()}
        content={this.makeContent()}
      />
    );
  }
}