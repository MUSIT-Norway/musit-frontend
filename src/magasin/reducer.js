import { toFrontend } from './mapper';
import * as types from './types';

const initialState = {
  root: {
    data: {}
  }
};

const storageUnitGridReducer = (state = initialState, action = {}) => {
  switch (action.type) {
  case types.LOAD_SEVERAL:
    return {
      ...state,
      loading: true
    };
  case types.LOAD_SEVERAL_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.result
    };
  case types.LOAD_SEVERAL_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  case types.LOAD_ONE_SUCCESS:
    return {
      ...state,
      root: {
        ...state.root,
        loading: false,
        loaded: true,
        data: toFrontend(action.result)
      }
    };
  case types.LOAD_ONE_FAIL:
    return {
      ...state,
      root: {
        ...state.root,
        loading: false,
        loaded: false,
        error: action.error
      }
    };
  case types.CLEAR_ROOT: {
    return {
      ...state,
      root: {}
    };
  }
  case types.DELETE:
    return {
      ...state,
      loading: true
    };
  case types.DELETE_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: []
    };
  case types.DELETE_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  default:
    return state;
  }
};

export default storageUnitGridReducer;