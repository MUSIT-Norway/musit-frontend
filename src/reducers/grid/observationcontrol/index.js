const LOAD = 'musit/observationcontrol/LOAD'
const LOAD_SUCCESS = 'musit/observationcontrol/LOAD_SUCCESS'
const LOAD_FAIL = 'musit/observationcontrol/LOAD_FAIL'
const LOAD_ACTOR = 'musit/observationcontrol/LOAD_ACTOR'
const LOAD_ACTOR_SUCCESS = 'musit/observationcontrol/LOAD_ACTOR_SUCCESS'
const LOAD_ACTOR_FAILURE = 'musit/observationcontrol/LOAD_ACTOR_FAILURE'

const initialState = { data: [] }

//
// const initialState = {
//   data: [
//     {
//       id: 1,
//       type: 'control',
//       doneDate: '01.01.1983',
//       types: { ControlTemperature: true,
//         ControlHypoxicAir: null,
//         ControlRelativeHumidity: null,
//         ControlCleaning: null,
//         ControlLightingCondition: null,
//         ControlAlcohol: true,
//         ControlGas: null,
//         ControlMold: null,
//         ControlPest: true,
//         envdata: null },
//       doneBy: 'Blablabla...',
//       registeredDate: '01.01.1983',
//       registeredBy: 'Blabla...'
//     },
//     {
//       id: 2,
//       type: 'observation',
//       doneDate: '01.01.1984',
//       types: { ControlTemperature: true,
//         ControlHypoxicAir: null,
//         ControlRelativeHumidity: null,
//         ControlCleaning: null,
//         ControlLightingCondition: true,
//         ControlAlcohol: false,
//         ControlGas: null,
//         ControlMold: null,
//         envdata: null },
//       doneBy: 'Blablabla...',
//       registeredDate: '01.01.1983',
//       registeredBy: 'Blabla...'
//     },
//     {
//       id: 3,
//       type: 'observation',
//       doneDate: '01.01.1984',
//       types: { ControlTemperature: true,
//         ControlHypoxicAir: true,
//         ControlRelativeHumidity: true,
//         ControlCleaning: true,
//         ControlLightingCondition: true,
//         ControlAlcohol: true,
//         ControlGas: true,
//         ControlMold: true,
//         envdata: true },
//       doneBy: 'Blablabla...',
//       registeredDate: '01.01.1983',
//       registeredBy: 'Blabla...'
//     },
//     {
//       id: 4,
//       type: 'control',
//       doneDate: '01.01.1984',
//       types: { ControlTemperature: false,
//         ControlHypoxicAir: false,
//         ControlRelativeHumidity: false,
//         ControlCleaning: false,
//         ControlLightingCondition: false,
//         ControlAlcohol: false,
//         ControlGas: false,
//         ControlMold: false,
//         envdata: false },
//       doneBy: 'Blablabla...',
//       registeredDate: '01.01.1983',
//       registeredBy: 'Blabla...'
//     },
//     {
//       id: 5,
//       type: 'observation',
//       doneDate: '01.01.1984',
//       types: { ControlTemperature: null,
//         ControlHypoxicAir: null,
//         ControlRelativeHumidity: null,
//         ControlCleaning: null,
//         ControlLightingCondition: null,
//         ControlAlcohol: true,
//         ControlGas: null,
//         ControlMold: null,
//         envdata: null },
//       doneBy: 'Blablabla...',
//       registeredDate: '01.01.1983',
//       registeredBy: 'Blabla...'
//     },
//     {
//       id: 6,
//       type: 'control',
//       doneDate: '01.01.1984',
//       types: { ControlTemperature: false
//          },
//       doneBy: 'Blablabla...',
//       registeredDate: '01.01.1983',
//       registeredBy: 'Blabla...'
//     }
//   ]
// }


const observationControlGridReducer = (state = initialState, action) => {
  switch (action.type) {
    case LOAD:
      return {
        ...state,
        loading: true,
        loaded: false
      };
    case LOAD_SUCCESS:
      return {
        ...state,
        loading: false,
        loaded: true,
        data: action.result
      };
    case LOAD_FAIL:
      return {
        ...state,
        loading: false,
        loaded: false,
        error: action.error
      }
    case LOAD_ACTOR:
      return {
        ...state,
        loading: true,
        loaded: false
      };
    case LOAD_ACTOR_SUCCESS:
      return {
        ...state,
        loading: false,
        loaded: true,
        data: state.data.map((e) => { return { ...e, doneName: action.result.fn } })
      };
    case LOAD_ACTOR_FAILURE:
      return {
        ...state,
        loading: false,
        loaded: false,
        error: action.error
      }
    default:
      return state;
  }
}

export const loadActor = (id) => {
  return {
    types: [LOAD_ACTOR, LOAD_ACTOR_SUCCESS, LOAD_ACTOR_FAILURE],
    promise: (client) => client.get(`/api/actor/v1/person/${id}`)
  }
}

export const loadControlsForNode = (id, callback) => {
  return {
    types: [LOAD, LOAD_SUCCESS, LOAD_FAIL],
    promise: (client) => client.get(`/api/event/v1/node/${id}/controls`),
    callback
  }
}
export const loadObservationsForNode = (id, callback) => {
  return {
    types: [LOAD, LOAD_SUCCESS, LOAD_FAIL],
    promise: (client) => client.get(`/api/event/v1/node/${id}/observations`),
    callback
  }
}

export const loadControlsAndObservationsForNode = (id, callback) => {
  return {
    types: [LOAD, LOAD_SUCCESS, LOAD_FAIL],
    promise: (client) => client.get(`/api/event/v1/node/${id}/controlsAndObservations`),
    callback
  }
}

export default observationControlGridReducer
