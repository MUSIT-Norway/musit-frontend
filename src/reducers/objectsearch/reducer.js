import { omit } from 'lodash';
import { getPath } from '../helper';
import MusitObject from '../../models/object';
import * as types from './constants';

const initialState = {
  data: {
    totalMatches: 0,
    matches: []
  },
  params: {
    currentPage: 1,
    perPage: 50
  },
  loaded: true
};

export default (state = initialState, action) => {
  switch(action.type) {
  case types.CHANGE_FIELD:
    return {
      ...state,
      params: {
        ...state.params,
        [action.field]: action.value
      }
    };
  case types.SEARCH_OBJECTS:
    return {
      ...omit(state, 'error'),
      loading: true,
      loaded: false,
      data: {
        totalMatches: 0,
        matches: []
      },
      params: {
        ...state.params,
        currentPage: action.page
      }
    };
  case types.SEARCH_OBJECTS_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      data: [],
      error: action.error
    };
  case types.SEARCH_OBJECTS_SUCCESS:
    return {
      ...omit(state, 'error'),
      loading: false,
      loaded: true,
      data: {
        totalMatches: action.result.totalMatches,
        matches: action.result.matches.map(data => {
          return new MusitObject({
            ...data,
            breadcrumb: getPath(data)
          });
        })
      }
    };
  default:
    return state;
  }
};