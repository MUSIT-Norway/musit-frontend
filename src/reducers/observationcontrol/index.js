const LOAD = 'musit/observationcontrol/LOAD'
const LOAD_SUCCESS = 'musit/observationcontrol/LOAD_SUCCESS'
const LOAD_FAIL = 'musit/observationcontrol/LOAD_FAIL'

export const initialState = {
  data: [{
    id: 1,
    eventType: 'controlEvent',
    doneBy: '',
    doneDate: '',
    registeredBy: '',
    registeredDate: '',
    observations: []
  },
    {
      id: 2,
      eventType: 'controlEvent',
      doneBy: '',
      doneDate: '',
      registeredBy: '',
      registeredDate: '',
      temperatureOK: false,
      inertAirOK: true,
      gasOK: true,
      lightConditionsOK: false,
      cleaningOK: false,
      alchoholOK: false,
      moldFungusOK: true,
      relativeHumidityOK: true,
      pestOK: false,
      storageUnit: false
    }, {
      id: 3,
      eventType: 'observationEvent',
      doneBy: '',
      doneDate: '',
      registeredBy: '',
      registeredDate: '',
      observations: []
    }, {
      id: 4,
      eventType: 'controlEvent',
      doneBy: '',
      doneDate: '',
      registeredBy: '',
      registeredDate: '',
      temperatureOK: true,
      inertAirOK: true,
      gasOK: false,
      lightConditionsOK: false,
      cleaningOK: true,
      alchoholOK: false,
      moldFungusOK: true,
      relativeHumidityOK: true,
      pestOK: true,
      storageUnit: true
    }, {
      id: 5,
      eventType: 'observationEvent',
      doneBy: '',
      doneDate: '',
      registeredBy: '',
      registeredDate: '',
      observations: []
    }]
}

const observationControlReducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case LOAD:
      return {
        ...state,
        loading: true,
        loaded: false,
        data: {}
      };
    case LOAD_SUCCESS:
      return {
        ...state,
        loading: false,
        loaded: true,
        data: action.result
      };
    case LOAD_FAIL:
      return {
        ...state,
        loading: false,
        loaded: false,
        data: action.error
      };
    default:
      return state;
  }
}

export default observationControlReducer;

export const loadObservationControl = (id) => {
  return {
    types: [LOAD, LOAD_SUCCESS, LOAD_FAIL],
    promise: (client) => client.get(`api/event/v1/event/${id}`)
  }
}
