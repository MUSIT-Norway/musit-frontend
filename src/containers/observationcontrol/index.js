
/*
 *  MUSIT is a museum database to archive natural and cultural history data.
 *  Copyright (C) 2016  MUSIT Norway, part of www.uio.no (University of Oslo)
 *
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License,
 *  or any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License along
 *  with this program; if not, write to the Free Software Foundation, Inc.,
 *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
import { I18n } from 'react-i18nify'
import { connect } from 'react-redux'
import { loadControlsAndObservationsForNode, loadActor } from '../../reducers/grid/observationcontrol'
import { loadRoot } from '../../reducers/storageunit/grid'
import ObservationControlGridShow from '../../components/observationcontrol'
import { createSelector } from 'reselect'
import orderBy from 'lodash/orderBy'

const getObservationControl = (state) => state.observationControlGrid.data

const getSortedObservationControl = createSelector(
    [ getObservationControl ],
    (observationControl) => orderBy(observationControl, ['doneDate', 'id'], ['desc', 'desc'])
)

const mapStateToProps = (state) => {
  return {
    translate: (key, markdown) => I18n.t(key, markdown),
    observationControlGridData: getSortedObservationControl(state),
    rootNode: state.storageGridUnit.root.data
  }
}

const mapDispatchToProps = (dispatch) => ({
  loadControlAndObservations: (id, callback) => {
    dispatch(loadControlsAndObservationsForNode(id, callback))
  },
  loadActorDetails: (data) => {
    dispatch(loadActor(data))
  },
  loadStorageObj: (id) => {
    dispatch(loadRoot(id))
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(ObservationControlGridShow)
