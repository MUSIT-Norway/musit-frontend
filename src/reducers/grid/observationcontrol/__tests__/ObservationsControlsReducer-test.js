import assert from 'assert';
import observationControlGridReducer from '../index';

import * as actions from '../../observationcontrol/index';
import reducer from '../../observationcontrol/index';
import Config from '../../../../config';
import request from 'superagent';
import nocker from 'superagent-nock';
const nock = nocker(request);

const initialState = {
  data: [
    { eventType: 'Observation',
    doneDate: '1990-11-11',
    doneBy: '2',
    'subEvents-parts': [
      { eventType: 'ObservationLightingCondition',
      lightingCondition: 'Veldig lyst idag',
      note: 'Skru ned lyset' },
      { eventType: 'ObservationAlcohol',
      condition: 'Mye alkohol',
      note: 'Tynn ut',
      volume: '93.5' },
      { eventType: 'ObservationPest',
      identification: 'Banafluer i fruktkurven',
      note: 'Eplecidereddik og Zalo',
      lifeCycles: [{ stage: 'Puppeskinn', number: '3' }, { stage: 'Larva', number: '3' }] }
    ] },
    { eventType: 'Control',
    doneDate: '1981-12-01',
    doneBy: '1',
    'subEvents-parts': [
      { eventType: 'ControlRelativeHumidity',
       OK: true },
      { eventType: 'ControlLightingCondition',
       OK: true },
      { eventType: 'ControlTemperature',
       OK: false },
      { eventType: 'ControlHypoxicAir',
       OK: false },
      { eventType: 'ControlAlcohol',
       OK: true }
    ]
   },
    { eventType: 'Observation',
    doneDate: '1994-02-21',
    doneBy: '2',
    'subEvents-parts': [
      { eventType: 'ObservationMold',
      mold: 'Muggent',
      note: 'Eddik' },
      { eventType: 'ObservationCleaning',
      cleaning: 'Urent',
      note: 'Her må der vaskes bedre' },
      { eventType: 'ObservationPest',
      identification: 'Sommerfugler i under takbjelken',
      note: 'Gasses',
      lifeCycles: [{ stage: 'Puppe', number: '1' }, { stage: 'Larva', number: '3' }] }
    ] },
    { eventType: 'Control',
    doneDate: '1891-12-01',
    doneBy: '2',
    'subEvents-parts': [
      { eventType: 'ControlCleaning',
       OK: true },
      { eventType: 'ControlHypoxicAir',
       OK: true },
      { eventType: 'ControlGass',
       OK: false },
      { eventType: 'ControlHypoxicAir',
       OK: false },
      { eventType: 'ControlPest',
       OK: true }
    ] }
  ]
};

describe('ObservationControlGridReducer', () => {

  it('Initial state is set', () => {
    const state = observationControlGridReducer(initialState, {});
    assert(state === initialState);
  });

  it('creates LOAD_ACTOR_SUCCESS when fetching data has been done', () => {
    const actorPostData = [1,2];
    const url = '/api/actor/v1/person/details';
    nock('http://localhost')
            .post(url, actorPostData)
            .reply(201, [
              {
                id: 1,
                fn: 'Jarle Stabell',
                dataportenId: 'jarle'
              },
              {
                id: 2,
                fn: 'Stein A. Olsen',
                dataportenId: 'stein'
              }
            ]);
    const store = mockStore();

    return store.dispatch(actions.loadActor([1, 2]))
            .then(() => {
              expect(store.getActions()).toMatchSnapshot();
            });
  });

  it('actor: no action', () => {
    expect(
            reducer(undefined, {})
        ).toMatchSnapshot();
  });

  it('actor: initial action', () => {
    expect(
            reducer(undefined, {
              type: actions.LOAD_ACTOR
            })
        ).toMatchSnapshot();
  });

  it('actor: success action', () => {
    expect(
            reducer(undefined, {
              type: actions.LOAD_ACTOR_SUCCESS,
              result: {
                someField: 1
              }
            })
        ).toMatchSnapshot();
  });

  it('actor: fail action', () => {
    expect(
            reducer(undefined, {
              type: actions.LOAD_ACTOR_FAILURE,
              error: Error('Some error in loadActor.')
            })
        ).toMatchSnapshot();
  });


  it('creates LOAD_SUCCESS when fetching data has been done', () => {
    const OutputSuccess = {
      data: [
                { eventType: 'Observation',
                    doneDate: '1990-11-11',
                    doneBy: '2',
                    'subEvents-parts': [
                        { eventType: 'ObservationLightingCondition',
                            lightingCondition: 'Veldig lyst idag',
                            note: 'Skru ned lyset' },
                        { eventType: 'ObservationAlcohol',
                            condition: 'Mye alkohol',
                            note: 'Tynn ut',
                            volume: '93.5' },
                        { eventType: 'ObservationPest',
                            identification: 'Banafluer i fruktkurven',
                            note: 'Eplecidereddik og Zalo',
                            lifeCycles: [{ stage: 'Puppeskinn', number: '3' }, { stage: 'Larva', number: '3' }] }
                    ] },
                { eventType: 'Control',
                    doneDate: '1981-12-01',
                    doneBy: '1',
                    'subEvents-parts': [
                        { eventType: 'ControlRelativeHumidity',
                            OK: true },
                        { eventType: 'ControlLightingCondition',
                            OK: true },
                        { eventType: 'ControlTemperature',
                            OK: false },
                        { eventType: 'ControlHypoxicAir',
                            OK: false },
                        { eventType: 'ControlAlcohol',
                            OK: true }
                    ]
                },
                { eventType: 'Observation',
                    doneDate: '1994-02-21',
                    doneBy: '2',
                    'subEvents-parts': [
                        { eventType: 'ObservationMold',
                            mold: 'Muggent',
                            note: 'Eddik' },
                        { eventType: 'ObservationCleaning',
                            cleaning: 'Urent',
                            note: 'Her må der vaskes bedre' },
                        { eventType: 'ObservationPest',
                            identification: 'Sommerfugler i under takbjelken',
                            note: 'Gasses',
                            lifeCycles: [{ stage: 'Puppe', number: '1' }, { stage: 'Larva', number: '3' }] }
                    ] },
                { eventType: 'Control',
                    doneDate: '1891-12-01',
                    doneBy: '2',
                    'subEvents-parts': [
                        { eventType: 'ControlCleaning',
                            OK: true },
                        { eventType: 'ControlHypoxicAir',
                            OK: true },
                        { eventType: 'ControlGass',
                            OK: false },
                        { eventType: 'ControlHypoxicAir',
                            OK: false },
                        { eventType: 'ControlPest',
                            OK: true }
                    ] }
      ]
    };
    const id = 1;
    const url = `${Config.magasin.urls.storagefacility.baseUrl(99)}/${id}/events`;
    nock('http://localhost')
            .get(url)
            .reply(200, OutputSuccess);
    const store = mockStore();

    return store.dispatch(actions.loadControlsAndObservationsForNode(1))
            .then(() => {
              expect(store.getActions()).toMatchSnapshot();
            });
  });

  it('observation control: no action', () => {
    expect(
            reducer(undefined, {})
        ).toMatchSnapshot();
  });

  it('observation control: initial action', () => {
    expect(
            reducer(undefined, {
              type: actions.LOAD
            })
        ).toMatchSnapshot();
  });

  it('observation control: success action', () => {
    expect(
            reducer(undefined, {
              type: actions.LOAD_SUCCESS,
              result: {
                someField: 1
              }
            })
        ).toMatchSnapshot();
  });

  it('observation control: fail action', () => {
    expect(
            reducer(undefined, {
              type: actions.LOAD_FAIL,
              error: Error('Some error to load observation control data.')
            })
        ).toMatchSnapshot();
  });
});
