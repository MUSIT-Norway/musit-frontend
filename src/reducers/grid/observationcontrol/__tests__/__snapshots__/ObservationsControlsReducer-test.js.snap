exports[`ObservationControlGridReducer actor: fail action 1`] = `
Object {
  "data": Array [],
  "error": [Error: Some error in loadActor.],
  "loaded": false,
  "loading": false
}
`;

exports[`ObservationControlGridReducer actor: initial action 1`] = `
Object {
  "data": Array [],
  "loaded": false,
  "loading": true
}
`;

exports[`ObservationControlGridReducer actor: no action 1`] = `
Object {
  "data": Array []
}
`;

exports[`ObservationControlGridReducer actor: success action 1`] = `
Object {
  "data": Array [],
  "loaded": true,
  "loading": false
}
`;

exports[`ObservationControlGridReducer creates LOAD_ACTOR_SUCCESS when fetching data has been done 1`] = `
Array [
  Object {
    "type": "musit/observationcontrol/LOAD_ACTOR"
  },
  Object {
    "result": Array [
      Object {
        "dataportenId": "jarle",
        "fn": "Jarle Stabell",
        "id": 1
      },
      Object {
        "dataportenId": "stein",
        "fn": "Stein A. Olsen",
        "id": 2
      }
    ],
    "type": "musit/observationcontrol/LOAD_ACTOR_SUCCESS"
  }
]
`;

exports[`ObservationControlGridReducer creates LOAD_SUCCESS when fetching data has been done 1`] = `
Array [
  Object {
    "type": "musit/observationcontrol/LOAD"
  },
  Object {
    "result": Object {
      "data": Array [
        Object {
          "doneBy": "2",
          "doneDate": "1990-11-11",
          "eventType": "Observation",
          "subEvents-parts": Array [
            Object {
              "eventType": "ObservationLightingCondition",
              "lightingCondition": "Veldig lyst idag",
              "note": "Skru ned lyset"
            },
            Object {
              "condition": "Mye alkohol",
              "eventType": "ObservationAlcohol",
              "note": "Tynn ut",
              "volume": "93.5"
            },
            Object {
              "eventType": "ObservationPest",
              "identification": "Banafluer i fruktkurven",
              "lifeCycles": Array [
                Object {
                  "number": "3",
                  "stage": "Puppeskinn"
                },
                Object {
                  "number": "3",
                  "stage": "Larva"
                }
              ],
              "note": "Eplecidereddik og Zalo"
            }
          ]
        },
        Object {
          "doneBy": "1",
          "doneDate": "1981-12-01",
          "eventType": "Control",
          "subEvents-parts": Array [
            Object {
              "OK": true,
              "eventType": "ControlRelativeHumidity"
            },
            Object {
              "OK": true,
              "eventType": "ControlLightingCondition"
            },
            Object {
              "OK": false,
              "eventType": "ControlTemperature"
            },
            Object {
              "OK": false,
              "eventType": "ControlHypoxicAir"
            },
            Object {
              "OK": true,
              "eventType": "ControlAlcohol"
            }
          ]
        },
        Object {
          "doneBy": "2",
          "doneDate": "1994-02-21",
          "eventType": "Observation",
          "subEvents-parts": Array [
            Object {
              "eventType": "ObservationMold",
              "mold": "Muggent",
              "note": "Eddik"
            },
            Object {
              "cleaning": "Urent",
              "eventType": "ObservationCleaning",
              "note": "Her m√• der vaskes bedre"
            },
            Object {
              "eventType": "ObservationPest",
              "identification": "Sommerfugler i under takbjelken",
              "lifeCycles": Array [
                Object {
                  "number": "1",
                  "stage": "Puppe"
                },
                Object {
                  "number": "3",
                  "stage": "Larva"
                }
              ],
              "note": "Gasses"
            }
          ]
        },
        Object {
          "doneBy": "2",
          "doneDate": "1891-12-01",
          "eventType": "Control",
          "subEvents-parts": Array [
            Object {
              "OK": true,
              "eventType": "ControlCleaning"
            },
            Object {
              "OK": true,
              "eventType": "ControlHypoxicAir"
            },
            Object {
              "OK": false,
              "eventType": "ControlGass"
            },
            Object {
              "OK": false,
              "eventType": "ControlHypoxicAir"
            },
            Object {
              "OK": true,
              "eventType": "ControlPest"
            }
          ]
        }
      ]
    },
    "type": "musit/observationcontrol/LOAD_SUCCESS"
  }
]
`;

exports[`ObservationControlGridReducer observation control: fail action 1`] = `
Object {
  "data": Array [],
  "error": [Error: Some error to load observation control data.],
  "loaded": false,
  "loading": false
}
`;

exports[`ObservationControlGridReducer observation control: initial action 1`] = `
Object {
  "data": Array [],
  "loaded": false,
  "loading": true
}
`;

exports[`ObservationControlGridReducer observation control: no action 1`] = `
Object {
  "data": Array []
}
`;

exports[`ObservationControlGridReducer observation control: success action 1`] = `
Object {
  "data": Object {
    "someField": 1
  },
  "loaded": true,
  "loading": false
}
`;
